modules/gapi/src/api/media.cpp

```diff
@@ -0,0 +1,41 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#include "precomp.hpp"
+#include <opencv2/gapi/media.hpp>
+
+struct cv::MediaFrame::Priv {
+    std::unique_ptr<IAdapter> adapter;
+};
+
+cv::MediaFrame::MediaFrame() {
+}
+
+cv::MediaFrame::MediaFrame(AdapterPtr &&ptr)
+    : m(new Priv{std::move(ptr)}) {
+}
+
+cv::GFrameDesc cv::MediaFrame::desc() const {
+    return m->adapter->meta();
+}
+
+cv::MediaFrame::View cv::MediaFrame::access(Access code) {
+    return m->adapter->access(code);
+}
+
+cv::MediaFrame::View::View(Ptrs&& ptrs, Strides&& strs, Callback &&cb)
+    : ptr   (std::move(ptrs))
+    , stride(std::move(strs))
+    , m_cb  (std::move(cb)) {
+}
+
+cv::MediaFrame::View::~View() {
+    GAPI_Assert(m_cb != nullptr);
```

Comment:
Oh this may be dangerous, as `GAPI_Assert` throws exceptions and `~View` could be called in context of an unwinding stack.